app/views/shared/_message_summary.html.haml

```diff
@@ -0,0 +1,14 @@
+- summarizer = MessageSummarizer.new(controller)
+- if summarizer.messages?
+  %li
+    %ul.nav.pull-right
+      %li.dropdown.pull-right
+        = link_to "Notices (#{summarizer.message_count})",
+          :notifications,
```

Comment:
Right. Clicking the "link" opens the drop-down. It's probably confusing to use "notices"/"notifications" to mean both the specific `Notification` and the collection of all types of message. I'm calling them, collectively, "messages" internally.